sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -193,6 +204,10 @@ private[sql] object PartitioningUtils {
         distinctPartColNames.size == 1,
         listConflictingPartitionColumns(pathsWithPartitionValues))
 
+      assert(
```

Comment:
Move this `assert` out of the `resolvePartitions`, since it's not so tight with the original purpose of this function, and we can avoid the code change